pkg/rid/cockroach/store_test.go

```diff
@@ -155,11 +150,92 @@ func TestGetVersion(t *testing.T) {
 	// TODO: remove the below checks when we have better schema management
 	require.Equal(t, "v2", semver.Major(version))
 
-	_, err = store.Queryable.ExecContext(ctx, `CREATE TABLE IF NOT EXISTS cells_subscriptions (id STRING PRIMARY KEY);`)
+	_, err = store.db.ExecContext(ctx, `CREATE TABLE IF NOT EXISTS cells_subscriptions (id STRING PRIMARY KEY);`)
 	require.NoError(t, err)
 
 	version, err = store.GetVersion(ctx)
 	require.NoError(t, err)
 	require.Equal(t, "v1", semver.Major(version))
+}
+
+func TestTransactor(t *testing.T) {
+	var (
+		ctx                  = context.Background()
+		store, tearDownStore = setUpStore(ctx, t)
+	)
+	require.NotNil(t, store)
+	defer tearDownStore()
+	subscription := subscriptionsPool[0].input
+	subscription1 := subscriptionsPool[1].input
+
+	txnCount := 0
+
+	err := store.InTxnRetrier(ctx, func(s2 repos.Repository) error {
+		if txnCount > 0 {
+			return errors.New("bail early")
+		}
+		txnCount++
+		subs, err := s2.SearchSubscriptions(ctx, subscription.Cells)
+
+		require.NoError(t, err)
+		require.Len(t, subs, 0)
+
+		_, err = s2.InsertSubscription(ctx, subscription)
+		require.NoError(t, err)
+		// Query outside the txn
+		_, err = store.InsertSubscription(ctx, subscription1)
+		require.NoError(t, err)
+
+		subs, err = s2.SearchSubscriptions(ctx, subscription.Cells)
+
+		require.NoError(t, err)
+		require.Len(t, subs, 1)
+		return err
+	})
+	require.NoError(t, err)
+	subs, err := store.SearchSubscriptions(ctx, subscription.Cells)
+	require.NoError(t, err)
+
+	require.Len(t, subs, 2)
+}
+
+func TestBasicTxn(t *testing.T) {
+	var (
+		ctx                  = context.Background()
+		store, tearDownStore = setUpStore(ctx, t)
+	)
+	require.NotNil(t, store)
+	defer tearDownStore()
+	subscription := subscriptionsPool[0].input
+	subscription1 := subscriptionsPool[1].input
+
+	tx, err := store.db.Begin()
+
+	s2 := *(store.SubscriptionStore)
+	s2.Queryable = tx
+
+	require.NotEqual(t, store.SubscriptionStore.Queryable, s2.Queryable)
+
+	subs, err := s2.SearchSubscriptions(ctx, subscription.Cells)
+
+	require.NoError(t, err)
+	require.Len(t, subs, 0)
+
+	_, err = s2.InsertSubscription(ctx, subscription)
+	require.NoError(t, err)
+	// Query outside the txn
+	_, err = store.InsertSubscription(ctx, subscription1)
+	require.NoError(t, err)
+
+	subs, err = s2.SearchSubscriptions(ctx, subscription.Cells)
```

Comment:
Shouldn't this fail since the database knows the transaction cannot be committed because the state of the database upon which the initial SearchSubscriptions was based is now invalid due to the Subscription insertion outside the transaction?  Or, at least tx.Commit() below should fail if this statement doesn't fail?

Reply:
ya i was confused by this initially as well. Did some digging into the definition of serializable isolation, and a simple way to think of it is as: 2 transactions may not execute in parralel if there is not an order in which their results would be the same had they executed sequentially.

So basically because I'm not doing a secondary read within that transaction. Going to rewrite the tests so that they fail and we catch this.